src/Stratis.Features.Unity3dApi/Unity3dApiSettings.cs

```diff
@@ -0,0 +1,146 @@
+            var apiUri = new Uri(apiHost);
+
+            // Find out which port should be used for the API.
+            int apiPort = config.GetOrDefault("unityapi_apiport", DefaultAPIPort, this.logger);
+
+            // If no port is set in the API URI.
+            if (apiUri.IsDefaultPort)
+            {
+                this.ApiUri = new Uri($"{apiHost}:{apiPort}");
+                this.ApiPort = apiPort;
+            }
+            // If a port is set in the -apiuri, it takes precedence over the default port or the port passed in -apiport.
+            else
+            {
+                this.ApiUri = apiUri;
+                this.ApiPort = apiUri.Port;
+            }
+
+            // Set the keepalive interval (set in seconds).
+            int keepAlive = config.GetOrDefault("unityapi_keepalive", 0, this.logger);
+            if (keepAlive > 0)
+            {
+                this.KeepaliveTimer = new Timer
+                {
+                    AutoReset = false,
+                    Interval = keepAlive * 1000
+                };
+            }
+        }
+
+        /// <summary>Prints the help information on how to configure the API settings to the logger.</summary>
+        /// <param name="network">The network to use.</param>
+        public static void PrintHelp(Network network)
+        {
+            var builder = new StringBuilder();
+            
+            builder.AppendLine($"-unityapi_enable=<bool>                    Use unity3d API. Defaults to false.");
+            builder.AppendLine($"-unityapi_apiuri=<string>                  URI to node's API interface. Defaults to '{ DefaultApiHost }'.");
+            builder.AppendLine($"-unityapi_apiport=<0-65535>                Port of node's API interface. Defaults to { network.DefaultAPIPort }.");
+            builder.AppendLine($"-unityapi_keepalive=<seconds>              Keep Alive interval (set in seconds). Default: 0 (no keep alive).");
+            builder.AppendLine($"-unityapi_usehttps=<bool>                  Use https protocol on the API. Defaults to false.");
+            builder.AppendLine($"-unityapi_certificatefilepath=<string>     Path to the certificate used for https traffic encryption. Defaults to <null>. Password protected files are not supported. On MacOs, only p12 certificates can be used without password.");
+
+            NodeSettings.Default(network).Logger.LogInformation(builder.ToString());
+        }
+
+        /// <summary>
+        /// Get the default configuration.
+        /// </summary>
+        /// <param name="builder">The string builder to add the settings to.</param>
+        /// <param name="network">The network to base the defaults off.</param>
+        public static void BuildDefaultConfigurationFile(StringBuilder builder, Network network)
+        {
+            builder.AppendLine("####Unity3d API Settings####");
+            builder.AppendLine($"#Enable unity3d api support");
+            builder.AppendLine($"#-unityapi_enable=");
```

Comment:
This is a nitpick, but the hyphen shouldn't be there in the config file

Reply:
fixed